### PR TITLE
Fix sudo perms when running tests

### PR DESCRIPTION
**Pull request by tssurya**
#### What this PR does and why is it needed
`
mount --make-rshared /run/user/1000/netns failed: "operation not permitted"
`

#### Which issue(s) this PR fixes

Fixes #

#### Special notes for reviewers
Rather than making all the tests there go into sudo mode I moved out the test that requires the testNS framework into its own package so that we can run the other tests properly downstream.

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

**Review by martinkennelly — DISMISSED** [review]

**Comment by jluhrsen** [comment]
@tssurya , can you also add this change `s/It/ovntest.OnSupportedPlatformsIt` in the following 5 places in
go-controller/pkg/node/secondary_node_network_controller_test.go

[line 124](https://github.com/ovn-org/ovn-kubernetes/blob/32f09a1ca2b3b67d65ccb57f46d60253ea048ced/go-controller/pkg/node/secondary_node_network_controller_test.go#L124)
[line 149](https://github.com/ovn-org/ovn-kubernetes/blob/32f09a1ca2b3b67d65ccb57f46d60253ea048ced/go-controller/pkg/node/secondary_node_network_controller_test.go#L149)
[line 173](https://github.com/ovn-org/ovn-kubernetes/blob/32f09a1ca2b3b67d65ccb57f46d60253ea048ced/go-controller/pkg/node/secondary_node_network_controller_test.go#L149)
[line 198](https://github.com/ovn-org/ovn-kubernetes/blob/32f09a1ca2b3b67d65ccb57f46d60253ea048ced/go-controller/pkg/node/secondary_node_network_controller_test.go#L149)
[line 228](https://github.com/ovn-org/ovn-kubernetes/blob/32f09a1ca2b3b67d65ccb57f46d60253ea048ced/go-controller/pkg/node/secondary_node_network_controller_test.go#L149)

**Comment by tssurya** [comment]
> @tssurya , can you also add this change `s/It/ovntest.OnSupportedPlatformsIt` in the following 5 places in go-controller/pkg/node/secondary_node_network_controller_test.go
> 
> [line 124](https://github.com/ovn-org/ovn-kubernetes/blob/32f09a1ca2b3b67d65ccb57f46d60253ea048ced/go-controller/pkg/node/secondary_node_network_controller_test.go#L124) [line 149](https://github.com/ovn-org/ovn-kubernetes/blob/32f09a1ca2b3b67d65ccb57f46d60253ea048ced/go-controller/pkg/node/secondary_node_network_controller_test.go#L149) [line 173](https://github.com/ovn-org/ovn-kubernetes/blob/32f09a1ca2b3b67d65ccb57f46d60253ea048ced/go-controller/pkg/node/secondary_node_network_controller_test.go#L149) [line 198](https://github.com/ovn-org/ovn-kubernetes/blob/32f09a1ca2b3b67d65ccb57f46d60253ea048ced/go-controller/pkg/node/secondary_node_network_controller_test.go#L149) [line 228](https://github.com/ovn-org/ovn-kubernetes/blob/32f09a1ca2b3b67d65ccb57f46d60253ea048ced/go-controller/pkg/node/secondary_node_network_controller_test.go#L149)

nice catch yup

> @tssurya , can you also add this change `s/It/ovntest.OnSupportedPlatformsIt` in the following 5 places in go-controller/pkg/node/secondary_node_network_controller_test.go
> 
> [line 124](https://github.com/ovn-org/ovn-kubernetes/blob/32f09a1ca2b3b67d65ccb57f46d60253ea048ced/go-controller/pkg/node/secondary_node_network_controller_test.go#L124) [line 149](https://github.com/ovn-org/ovn-kubernetes/blob/32f09a1ca2b3b67d65ccb57f46d60253ea048ced/go-controller/pkg/node/secondary_node_network_controller_test.go#L149) [line 173](https://github.com/ovn-org/ovn-kubernetes/blob/32f09a1ca2b3b67d65ccb57f46d60253ea048ced/go-controller/pkg/node/secondary_node_network_controller_test.go#L149) [line 198](https://github.com/ovn-org/ovn-kubernetes/blob/32f09a1ca2b3b67d65ccb57f46d60253ea048ced/go-controller/pkg/node/secondary_node_network_controller_test.go#L149) [line 228](https://github.com/ovn-org/ovn-kubernetes/blob/32f09a1ca2b3b67d65ccb57f46d60253ea048ced/go-controller/pkg/node/secondary_node_network_controller_test.go#L149)

@jluhrsen hmm why is that? those things don't really do the host-level stuff.
Is it because I am using TestNS inside the beforeeach?

**Comment by jluhrsen** [comment]
> > @tssurya , can you also add this change `s/It/ovntest.OnSupportedPlatformsIt` in the following 5 places in go-controller/pkg/node/secondary_node_network_controller_test.go
> > [line 124](https://github.com/ovn-org/ovn-kubernetes/blob/32f09a1ca2b3b67d65ccb57f46d60253ea048ced/go-controller/pkg/node/secondary_node_network_controller_test.go#L124) [line 149](https://github.com/ovn-org/ovn-kubernetes/blob/32f09a1ca2b3b67d65ccb57f46d60253ea048ced/go-controller/pkg/node/secondary_node_network_controller_test.go#L149) [line 173](https://github.com/ovn-org/ovn-kubernetes/blob/32f09a1ca2b3b67d65ccb57f46d60253ea048ced/go-controller/pkg/node/secondary_node_network_controller_test.go#L149) [line 198](https://github.com/ovn-org/ovn-kubernetes/blob/32f09a1ca2b3b67d65ccb57f46d60253ea048ced/go-controller/pkg/node/secondary_node_network_controller_test.go#L149) [line 228](https://github.com/ovn-org/ovn-kubernetes/blob/32f09a1ca2b3b67d65ccb57f46d60253ea048ced/go-controller/pkg/node/secondary_node_network_controller_test.go#L149)
> > @tssurya , can you also add this change `s/It/ovntest.OnSupportedPlatformsIt` in the following 5 places in go-controller/pkg/node/secondary_node_network_controller_test.go
> > [line 124](https://github.com/ovn-org/ovn-kubernetes/blob/32f09a1ca2b3b67d65ccb57f46d60253ea048ced/go-controller/pkg/node/secondary_node_network_controller_test.go#L124) [line 149](https://github.com/ovn-org/ovn-kubernetes/blob/32f09a1ca2b3b67d65ccb57f46d60253ea048ced/go-controller/pkg/node/secondary_node_network_controller_test.go#L149) [line 173](https://github.com/ovn-org/ovn-kubernetes/blob/32f09a1ca2b3b67d65ccb57f46d60253ea048ced/go-controller/pkg/node/secondary_node_network_controller_test.go#L149) [line 198](https://github.com/ovn-org/ovn-kubernetes/blob/32f09a1ca2b3b67d65ccb57f46d60253ea048ced/go-controller/pkg/node/secondary_node_network_controller_test.go#L149) [line 228](https://github.com/ovn-org/ovn-kubernetes/blob/32f09a1ca2b3b67d65ccb57f46d60253ea048ced/go-controller/pkg/node/secondary_node_network_controller_test.go#L149)
> 
> @jluhrsen hmm why is that? those things don't really do the host-level stuff. Is it because I am using TestNS inside the beforeeach?

I have no idea. I don't know these tests at all. I just used your change as a shot in the dark for each of the 5 tests that were
failing in d/s unit tests and it fixed it.

[without ovntest.OnSupportedPlatformsIt](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_ovn-kubernetes/2290/pull-ci-openshift-ovn-kubernetes-master-unit/1833906047643815936)
[with ovntest.OnSupportedPlatformsIt](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_ovn-kubernetes/2290/pull-ci-openshift-ovn-kubernetes-master-unit/1833962792403603456)

**Comment by tssurya** [comment]
> I have no idea. I don't know these tests at all. 

ack can you help me confirm that its because of the beforeeach so that we can get that idea correctly? :)

> I just used your change as a shot in the dark for each of the 5 tests that were failing in d/s unit tests and it fixed it.

ok I will update this PR under the assumption that its because of the BeforeEach I am doing. I would prefer we can confirm that assumption.

**Comment by jluhrsen** [comment]
> > I have no idea. I don't know these tests at all.
> 
> ack can you help me confirm that its because of the beforeeach so that we can get that idea correctly? :)
> 
> > I just used your change as a shot in the dark for each of the 5 tests that were failing in d/s unit tests and it fixed it.
> 
> ok I will update this PR under the assumption that its because of the BeforeEach I am doing. I would prefer we can confirm that assumption.

@tssurya , I think I validated your thoughts about the BeforeEach()

I removed the BeforeEach() in [this PR](https://github.com/openshift/ovn-kubernetes/pull/2290) and added that init stuff directly in to the test case and it passed. Then I made [this test
PR ](https://github.com/openshift/ovn-kubernetes/pull/2301 )which just runs the single test case with the BeforeEach() back and it failed with the `permission denied` error.

not sure what you want to do next, but if I can help please let me know

**Comment by tssurya** [comment]
> > > I have no idea. I don't know these tests at all.
> > 
> > 
> > ack can you help me confirm that its because of the beforeeach so that we can get that idea correctly? :)
> > > I just used your change as a shot in the dark for each of the 5 tests that were failing in d/s unit tests and it fixed it.
> > 
> > 
> > ok I will update this PR under the assumption that its because of the BeforeEach I am doing. I would prefer we can confirm that assumption.
> 
> @tssurya , I think I validated your thoughts about the BeforeEach()
> 
> I removed the BeforeEach() in [this PR](https://github.com/openshift/ovn-kubernetes/pull/2290) and added that init stuff directly in to the test case and it passed. Then I made [this test PR ](https://github.com/openshift/ovn-kubernetes/pull/2301)which just runs the single test case with the BeforeEach() back and it failed with the `permission denied` error.
> 
> not sure what you want to do next, but if I can help please let me know

yup that's it, thanks. so that's the approach I took to fix in this PR, so if you review and if this fix works, and leave an lgtm I can merge this.
waiting for reviews at this stage.

**Comment by tssurya** [comment]
latest push is just rebase on master, once CI runs will merge this since @jluhrsen has given lgtm for this

**Review by jluhrsen — APPROVED** [review]

**Pull request merged**
